scripts/Interface/ui/dataprocessorinterface/data_processor_gui.py

```diff
@@ -0,0 +1,221 @@
+        self._create_action(MantidQt.MantidWidgets.DataProcessorDeleteRowCommand(self.data_processor_table), self.menuEdit)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorDeleteGroupCommand(self.data_processor_table), self.menuEdit)
+
+        # Actions that go in the 'File' menu
+        self._create_action(MantidQt.MantidWidgets.DataProcessorOpenTableCommand(self.data_processor_table), self.menuFile, workspace_list)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorNewTableCommand(self.data_processor_table), self.menuFile)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorSaveTableCommand(self.data_processor_table), self.menuFile)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorSaveTableAsCommand(self.data_processor_table), self.menuFile)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorImportTableCommand(self.data_processor_table), self.menuFile)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorExportTableCommand(self.data_processor_table), self.menuFile)
+        self._create_action(MantidQt.MantidWidgets.DataProcessorOptionsCommand(self.data_processor_table), self.menuFile)
+
+    def _create_action(self, command, menu, workspace_list = None):
+        """
+        Create an action from a given DataProcessorCommand and add it to a given menu
+        A 'workspace_list' can be provided but it is only intended to be used with DataProcessorOpenTableCommand.
+        It refers to the list of table workspaces in the ADS that could be loaded into the widget. Note that only
+        table workspaces with an appropriate number of columns and column types can be loaded.
+        """
+
+        if (workspace_list is not None and command.name() == "Open Table"):
+            submenu = QtGui.QMenu(command.name(), self)
+            submenu.setIcon(QtGui.QIcon(command.icon()))
+
+            for ws in workspace_list:
+                ws_command = MantidQt.MantidWidgets.DataProcessorWorkspaceCommand(self.data_processor_table, ws)
+                action = QtGui.QAction(QtGui.QIcon(ws_command.icon()), ws_command.name(), self)
+                action.triggered.connect(lambda: self._connect_action(ws_command))
+                submenu.addAction(action)
+
+            menu.addMenu(submenu)
+        else:
+            action = QtGui.QAction(QtGui.QIcon(command.icon()), command.name(), self)
+            action.setShortcut(command.shortcut())
+            action.setStatusTip(command.tooltip())
+            action.triggered.connect(lambda: self._connect_action(command))
+            menu.addAction(action)
+
+    def _connect_action(self, command):
+        """
+        Executes an action
+        """
+        command.execute()
+
+    def _run_python_code(self, text):
+        """
+        Re-emits 'runPytonScript' signal
+        """
+        mantidplot.runPythonScript(text, True)
```

Comment:
Looks good. We've already discussed the possibility of adding another  toy example.